cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/MapReduceProgramWorkflowRunner.java

```diff
@@ -51,8 +52,8 @@
    */
   @Override
   public Callable<RuntimeContext> create(String name) {
-
-    final MapReduceSpecification mapReduceSpec = workflowSpec.getMapReduce().get(name);
+    ApplicationSpecification spec = workflowProgram.getSpecification();
```

Comment:
Yes. agree. Changed the method name to the getApplicationSpecification. Since it was used in almost 35 different files those files got added to this PR as well :(